scripts/create_gitlab_snapshot.py

```diff
@@ -0,0 +1,60 @@
+import datetime
+import logging
+from more_itertools import (
+    one,
+)
+
+from azul import (
+    config,
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+from azul.terraform import (
+    _adjust_name_tag,
+    _tags,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def create_snapshot():
+    require(config.terraform_component == 'gitlab',
+            'Please _select the GitLab environment for the targeted deployment')
+    filters_ = [
+        {
+            'Name': 'key-name',
+            'Values': ['azul-gitlab']
+        }
+    ]
+    instance = one([
+        one(reservation['Instances'])
+        for reservation in aws.ec2.describe_instances(Filters=filters_)['Reservations']
+    ])
+    assert instance['State']['Name'] == 'stopped', instance['State']
+    resource_tags = [
+        {'Key': k, 'Value': v}
+        for k, v in _adjust_name_tag('aws_gitlab',
+                                     _tags('gitlab', name='azul_gitlab')).items()
+    ]
+    operator_name = aws.sts.get_caller_identity()['UserId'].split(':')[1]
+    assert operator_name
+    resource_tags.append({'Key': 'created-by', 'Value': operator_name})
+    date = datetime.datetime.now().strftime('%Y-%m-%d %H:%M')
+    aws.ec2.create_snapshots(Description=f'{date} Gitlab Update',
+                             InstanceSpecification=dict(InstanceId=instance['InstanceId'],
+                                                        ExcludeBootVolume=False),
+                             TagSpecifications=[dict(Tags=resource_tags,
+                                                     ResourceType='snapshot')])
+    logger.info('Created snapshots for %s, check progress within the AWS Console',
+                instance['KeyName'])
```

Comment:
Honestly, if I have to use the console anyways, why bother with this script? 

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Snapshot.wait_until_completed

Reply:
Mainly to ensure snapshot tagging is consistent, manually creating 2 snapshots for every instance and setting all the tags leaves room for errors.

I added both wait for shutdown and wait for snapshot to the script, so the operator does not need to use the AWS console.